utils/ixdtf/src/parsers/mod.rs

```diff
@@ -0,0 +1,327 @@
+            set.calendar
+        } else {
+            None
+        };
+
+        self.cursor.close()?;
+
+        Ok(IsoParseRecord {
+            date: DateRecord {
+                year: 0,
+                month: month_day.0,
+                day: month_day.1,
+            },
+            time: None,
+            tz: None,
+            calendar,
+        })
+    }
+
+    /// Parses the source as a [Time string][temporal-time].
+    ///
+    /// This method will parse as an annotated time string or annotated date time string.
+    ///
+    /// [temporal-time]: https://tc39.es/proposal-temporal/#prod-TemporalTimeString
+    pub fn parse_time(&mut self) -> ParserResult<IsoParseRecord> {
+        match time::parse_annotated_time_record(&mut self.cursor) {
+            Ok(None) => {
+                self.cursor.pos = 0;
+                datetime::parse_annotated_date_time(DateTimeFlags::TIME_REQ, &mut self.cursor)
+            }
+            Ok(Some(result)) => Ok(result),
+            Err(err) => Err(err),
+        }
+    }
+
+    /// Parses the source as a Time Zone string.
+    pub fn parse_time_zone(&mut self) -> ParserResult<TimeZone> {
+        time_zone::parse_time_zone(&mut self.cursor)
+    }
+}
+
+/// A parser for Duration strings.
+///
+/// # Exmaple
+///
+/// ```
+/// use ixdtf::parsers::{IsoDurationParser, records::DurationParseRecord };
+///
+/// let duration_str = "P1Y2M3W1D";
+///
+/// let result = IsoDurationParser::new(duration_str).parse().unwrap();
+///
+/// assert_eq!(result.years, 1);
+/// assert_eq!(result.months, 2);
+/// assert_eq!(result.weeks, 3);
+/// assert_eq!(result.days, 1);
+///
+/// ```
+#[derive(Debug)]
+pub struct IsoDurationParser {
+    cursor: Cursor,
+}
+
+impl IsoDurationParser {
+    /// Creates a new `IsoDurationParser` from a target `&str`.
+    pub fn new(target: &str) -> Self {
+        Self {
+            cursor: Cursor::new(target),
+        }
+    }
+
+    /// Parse the contents of this `IsoDurationParser` into a `DurationParseRecord`.
+    pub fn parse(&mut self) -> ParserResult<DurationParseRecord> {
+        parse_duration(&mut self.cursor)
+    }
+}
+
+// ==== Mini cursor implementation for Iso8601 targets ====
+
+/// `Cursor` is a small cursor implementation for parsing Iso8601 grammar.
+#[derive(Debug)]
+pub(crate) struct Cursor {
+    pos: u32,
+    source: Vec<char>,
+}
+
+impl Cursor {
+    /// Create a new cursor from a source `String` value.
+    #[must_use]
+    pub(crate) fn new(source: &str) -> Self {
+        Self {
+            pos: 0,
+            source: source.chars().collect(),
+        }
+    }
+
+    /// Returns a string value from a slice of the cursor.
+    fn slice(&self, start: u32, end: u32) -> String {
+        let slice = self.source.get(start as usize..end as usize);
+        let Some(slice) = slice else {
+            return String::default();
+        };
+        slice.iter().collect()
+    }
+
+    /// Get current position
+    const fn pos(&self) -> u32 {
```

Comment:
Question: why `u32` instead of `usize`? Seems not necessary to restrict it. These strings should definitely not be that long, but in general it's better for style and security to stick with the native size types. ICU4C has gotten in a little bit of trouble for using `i32` instead of `size_t`, for example.